airflow/models/mappedoperator.py

```diff
@@ -660,13 +659,11 @@ def expand_mapped_task(self, run_id: str, *, session: Session) -> Tuple[Sequence
 
         for index in indexes_to_map:
             # TODO: Make more efficient with bulk_insert_mappings/bulk_save_mappings.
-            # TODO: Change `TaskInstance` ctor to take Operator, not BaseOperator
-            ti = TaskInstance(self, run_id=run_id, map_index=index, state=state)  # type: ignore
+            ti = TaskInstance(self, run_id=run_id, map_index=index, state=state)
```

Comment:
Not related to the fix, but we already made the argument type `Operator` long ago.